crates/nu-parser/src/parse_keywords.rs

```diff
@@ -629,8 +630,8 @@ fn parse_def_inner(
                         expr: Expr::Closure(block_id),
                         ..
                     } => {
-                        let block = working_set.get_block_mut(*block_id);
-                        block.signature = Box::new(sig.clone());
+                        compile_block_with_id(working_set, *block_id);
+                        working_set.get_block_mut(*block_id).signature = Box::new(sig.clone());
```

Comment:
I saw your notes in the PR description, but I don't get it. I think we can defer this to the compiling of its parent block as long as the block id is accessible there.

Reply:
I see, thanks. There seems to be much more work required than I thought to totally separate parsing from compiling then.

I’m a bit obsessed with that because I’m thinking of many tasks that only interact with the AST, nu-lsp, reedline highlighting, ast command…

Perhaps compiling is not the performance bottleneck at all, just feels more mentally hygienic to do so.

If you could see a practical route to achieve that goal, maybe we can add some TODO notes somewhere.

Reply:
According to my test, this single line change helps to eliminate the dichotomy of `export-env` parsing.

<img width="847" height="261" alt="image" src="https://github.com/user-attachments/assets/26b6113b-d454-49bb-b3ee-695d36c129b8" />

Would you like to have another PR to review it separately, or merge that change in this one?